File: frontend/web/src/api/tools.ts
import { apiFetch } from "./client";

export type ToolCatalogEntry = {
  name: string;
  description: string;
  input_schema: unknown;
  built_in: boolean;
};

export type ToolsListResponse = {
  items: ToolCatalogEntry[];
};

export async function listTools(): Promise<ToolsListResponse> {
  return apiFetch<ToolsListResponse>("/api/tools");
}

export const toolKeys = {
  all: ["tools"] as const,
};

File: crates/xvision-engine/src/api/tools.rs
use serde::Serialize;
use serde_json::Value;

use crate::tools::built_in_tool_descriptors;

#[derive(Debug, Clone, Serialize)]
pub struct ToolCatalogEntry {
    pub name: String,
    pub description: String,
    pub input_schema: Value,
    pub built_in: bool,
}

pub async fn list_tools() -> Vec<ToolCatalogEntry> {
    built_in_tool_descriptors()
        .into_iter()
        .map(|descriptor| ToolCatalogEntry {
            name: descriptor.name,
            description: descriptor.description,
            input_schema: descriptor.input_schema,
            built_in: true,
        })
        .collect()
}
